codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -195,14 +195,26 @@ private FlowState connectionState() {
     }
 
     private static FlowState state(Http2Stream stream) {
-        checkNotNull(stream, "stream");
-        return stream.getProperty(FlowState.class);
+        if (stream instanceof FastPropertyAccess) {
```

Comment:
> instanceof actually is for free

Do you have a reference for this?  I'm thinking nothing is free...if JIT is optimizing it away because `checkcast` will handle it then there is cost associated with that.  The `instanceof` is also a byte code operation which seems to be very similar to the `checkcast` byte code operation.